lib/painter.dart

```diff
@@ -143,14 +148,30 @@ class _PathHistory {
     _inDrag = false;
   }
 
-  void draw(Canvas canvas, Size size) {
+  void draw(Canvas canvas, Size size, Image image) {
```

Comment:
I am a bit concerned about potential performance issues of this approach.

Because we cache path data (so its possible to undo), performance is worse with every new stroke.
That issue should be fixed some day, but it required much work due to advanced complexity.

Approach you introduced brings another weight to #draw method. I think that it would be much better if we just use flutter Stack widget to place background image under painting canvas.